### ISSUE TITLE
What am I doing wrong here?

### ISSUE THREAD

**Issue by TyrfingMjolnir** [lukas2511/dehydrated#431]
What am I doing wrong here?
```
$ ./dehydrated -c
# INFO: Using main config file /usr/local/etc/dehydrated/config
Processing domain.tld with alternative names: www.domain.tld
 + Signing domains...
 + Generating private key...
 + Generating signing request...
 + Requesting challenge for domain.tld...
 + Requesting challenge for www.domain.tld...
 + Responding to challenge for domain.tld...
 + Responding to challenge for www.domain.tld...
 + Challenge is valid!
 + Requesting certificate...
  + ERROR: An error occurred while sending post-request to https://acme-v01.api.letsencrypt.org/acme/new-cert (Status 403)

Details:
{
  "type": "urn:acme:error:unauthorized",
  "detail": "Error creating new cert :: authorizations for these names not found or expired: domain.tld",
  "status": 403
}
```

```
$ cat /usr/local/etc/dehydrated/domains.txt
domain.tld www.domain.tld
```

**Comment by txr13** [comment]
To my eye, that looks like the challenge for domain.tld wasn't validated...

Can you please post the results of `dehydrated --version` for us?

**Comment by txr13** [comment]
@lukas2511 Looking through the code, I think we may have a bug?

In lines 606-610, we loop while `reqstatus` is still pending. In lines 621-625, we check if `reqstatus` is valid, and if not, we call the hook script... but we don't error out otherwise. Which means we move to validating the next SAN in the list.

Once we're done with all the challenges, we move on. In line 635, we check again that `reqstatus` is valid, but that's really only checking whether the _last_ SAN validation succeeded.

**Comment by TyrfingMjolnir** [comment]
# ./dehydrated --version
# INFO: Using main config file /usr/local/etc/dehydrated/config
Dehydrated by Lukas Schauer
https://dehydrated.de

Dehydrated version: 0.4.0
GIT-Revision: 0be0ab083f290afbc757b8388a80df458ddfd33c

cat: /etc/issue: No such file or directory
OS:
Used software:
 bash: 4.3.30(1)-release
 sed: sed (GNU sed) 4.4
 curl: curl 7.53.1
 awk: GNU Awk 4.1.4, API: 1.1 (GNU MPFR 3.1.5, GNU MP 6.1.2)
 grep: grep (GNU grep) 2.27
 mktemp: mktemp (GNU coreutils) 8.26
 diff: diff (GNU diffutils) 3.5
 openssl: OpenSSL 1.0.2k  26 Jan 2017

# uname -a
SunOS reverseproxy01 5.11 joyent_20170913T233706Z i86pc i386 i86pc Solaris

**Comment by TyrfingMjolnir** [comment]
What I did wrong was to mount the location at www.domain.tld and did not assign domain.tld to anything.

**Comment by lukas2511** [comment]
@txr13 mh, you are right, that is a bug and i was able to reproduce it easily. i think a simple break should fix that, see f838d93 for my fix.
it would also be possible to move the cleanup routine up, in that case it may even be simplified since it wouldn't need the loop over the invalid challenge files but only delete the current one, but maybe i'm forgetting something...
